validation/pipeline/qase/qase.go

```diff
@@ -0,0 +1,174 @@
+}
+
+// GetAllTestCases retrieves all Test Cases within a specified Qase Project and returns a slice of them
+func GetAllTestCases(project string) ([]qase.TestCase, error) {
+	logrus.Debugln("Retrieving test cases from Qase for project id: ", project)
+	testCases := []qase.TestCase{}
+	var numOfTestCases int32 = 1
+	offSetCount := 0
+	for numOfTestCases > 0 {
+		offset := optional.NewInt32(int32(offSetCount))
+		localVarOptionals := &qase.CasesApiGetCasesOpts{
+			Offset: offset,
+		}
+		tempResult, _, err := Client.CasesApi.GetCases(context.TODO(), project, localVarOptionals)
+		if err != nil {
+			return nil, err
+		}
+
+		testCases = append(testCases, tempResult.Result.Entities...)
+		numOfTestCases = tempResult.Result.Count
+		offSetCount += 10
+	}
+
+	return testCases, nil
+}
+
+// GetTestSuite retrieves a Test Suite by name within a specified Qase Project if it exists
+func GetTestSuite(project, suite string) (qase.Suite, error) {
+	logrus.Debugf("Getting test suite \"%s\" in project %s\n", suite, project)
+	localVarOptionals := &qase.SuitesApiGetSuitesOpts{
+		FiltersSearch: optional.NewString(suite),
+	}
+	qaseSuites, _, err := Client.SuitesApi.GetSuites(context.TODO(), project, localVarOptionals)
+	if err != nil {
+		return qase.Suite{}, err
+	}
+
+	for _, qaseSuite := range qaseSuites.Result.Entities {
+		if qaseSuite.Title == suite {
+			return qaseSuite, nil
+		}
+	}
+	return qase.Suite{}, fmt.Errorf("test suite \"%s\" not found in project %s", suite, project)
+}
+
+// CreateTestSuite retrieves a Test Suite by name within a specified Qase Project if it exists
+func CreateTestSuite(project, suite string) (int64, error) {
+	logrus.Debugf("Creating test suite \"%s\" in project %s\n", suite, project)
+	suiteBody := qase.SuiteCreate{Title: suite}
+	resp, _, err := Client.SuitesApi.CreateSuite(context.TODO(), suiteBody, project)
+	if err != nil {
+		return 0, fmt.Errorf("failed to create test suite: \"%s\". Error: %v", suite, err)
+	}
+	return resp.Result.Id, nil
+}
+
+// UploadTests either creates new Test Cases and their associated Suite or updates them if they already exist
+func UploadTests(project string, testCases []TestCase) error {
+	for _, tc := range testCases {
+		logrus.Info("Uploading test case:\n\tProject: ", project, "\n\tTitle: ", tc.Title, "\n\tDescription: ", tc.Description, "\n\tSuiteId: ", tc.SuiteId, "\n\tAutomation: ", tc.Automation, "\n\tSteps: ", tc.Steps)
+
+		existingCase, err := GetTestCase(project, tc.Title)
+		if err == nil {
+			var qaseTest qase.TestCaseUpdate
+			qaseTest.Title = tc.Title
+			qaseTest.Description = tc.Description
+			qaseTest.SuiteId = tc.SuiteId
+			qaseTest.Automation = tc.Automation
+			for _, step := range tc.Steps {
+				var qaseSteps qase.TestCaseUpdateSteps
+				qaseSteps.Action = step.Action
+				qaseSteps.ExpectedResult = step.ExpectedResult
+				qaseSteps.Data = step.InputData
+				qaseSteps.Position = step.Position
+				qaseTest.Steps = append(qaseTest.Steps, qaseSteps)
```

Comment:
same idea for qaseSteps as for qaseTest

Reply:
in fact, this whole thing can probably be broken out into its own helper function since its shared between the if/else here and looks like it would generally be helpful on its own.